test/test-esm.js

```diff
@@ -0,0 +1,28 @@
+let t;
+
+try {
+  t = require('tap')
+} catch (e) {
+  t = require('../');
+  t.equals(e.code, 'MODULE_NOT_FOUND')
+  t.end();
+  return
+}
+
+const {spawn} = require('child_process')
```

Comment:
I think we should use `t.spawn` instead of child_process.spawn.  This will cause tap output from `test.mjs` to be processed as a sub-test.